src/analysis/souffle/tags.dl

```diff
@@ -0,0 +1,41 @@
+//-----------------------------------------------------------------------------
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-----------------------------------------------------------------------------
+
+// Definition of the base tag types that are used by this dataflow analysis.
+
+#ifndef SRC_ANALYSIS_SOUFFLE_TAGS_DL_
+#define SRC_ANALYSIS_SOUFFLE_TAGS_DL_
+
+// A symbol representing a taint tag. SecrecyTags indicate that the data upon which they are
+// annotated potentially contain information that should be handled with some sort of special care
+// and should not be exposed to some outside group. The conservative approximation for SecrecyTags
+// is, when you are in doubt whether a particular datum should have this tag, annotate it. This
+// causes these tags to spread like taint.
+.type SecrecyTag <: symbol
+
+// A symbol representing a guarantee tag. IntegrityTags indicate that the data upon which they are
+// annotated have some critical property that handlers must take care to preserve. The conservative
+// approximation for IntegrityTags is, when you are in doubt whether a particular datum should have
+// this tag, do not annotate it. This causes these tags to disappear unless carefully handled.
+.type IntegrityTag <: symbol
+
+// This is a tag that a user or some principal claims is present upon some accessPath. This
+// introduces *both* a secrecy and integrity tag upon that accessPath, which will each
+// independently spread through the graph from that point according to their different behavior
+// patterns.
+.type ClaimedTag <: symbol
```

Comment:
What's this for? Is it meant to distinguish between the tags that were explicitly given by some programmer from the tags that are inferred?